### PR TITLE
CON-3334 Display change of provider history

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/ApprenticeshipDetailsRequestViewModel.cs [hunk 1]
```diff
@@ -34,21 +35,17 @@ public class ApprenticeshipDetailsRequestViewModel : IAuthorizationContextModel
...
         public string HashedNewApprenticeshipId { get; set; }
-        public bool IsContinuation { get; set; }
-        public string HashedPreviousApprenticeshipId { get; set; }
         public bool ShowChangeTrainingProviderLink => ((ApprenticeshipStatus == ApprenticeshipStatus.Stopped ||
                                                        ApprenticeshipStatus == ApprenticeshipStatus.Paused ||
                                                        ApprenticeshipStatus == ApprenticeshipStatus.Live ||
                                                        ApprenticeshipStatus == ApprenticeshipStatus.WaitingToStart) &&
                                                        !HasPendingChangeOfProviderRequest &&
                                                        !HasPendingChangeOfEmployerRequest &&
-                                                       !(HasApprovedChangeOfEmployerRequest && !IsContinuation) &&
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/ApprenticeshipDetailsRequestViewModel.cs:45]: has this condition been correctly remove, meaning it was being applied incorrectly previously?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapper.cs [hunk 2]
```diff
@@ -85,26 +86,35 @@ public async Task<ApprenticeshipDetailsRequestViewModel> Map(ApprenticeshipDetai
                     EmployerReference = apprenticeship.EmployerReference,
                     CohortReference = _encodingService.Encode(apprenticeship.CohortId, EncodingType.CohortReference),
                     EnableEdit = enableEdit,
-                    CanEditStatus = (apprenticeship.Status == ApprenticeshipStatus.Live || 
-                                     apprenticeship.Status == ApprenticeshipStatus.WaitingToStart || 
-                                     apprenticeship.Status == ApprenticeshipStatus.Paused),                    
+                    CanEditStatus = (apprenticeship.Status == ApprenticeshipStatus.Live ||
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapper.cs:89]: ApprenticeshipStatus is set a little further up this mapper, could this logic not just be a property getter? I know this is not something we have changed here. The whitespace changes just highlighted it to me

**Review by cofaulco — COMMENTED** [review]
couple of comments

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/ApprenticeshipDetailsRequestViewModel.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/ApprenticeshipDetailsRequestViewModel.cs:45]: Yes - the next line is using HashedNewApprenticeshipId which is now correctly obtained for both ChangeOfEmployer and ChangeOfProvider - this was a bug identified in the previous migration work and an AC was added to this ticket to fix it as the testing in in the same area

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapper.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapper.cs:89]: Done

**Review by cofaulco — APPROVED** [review]

**Pull request merged**